### PR TITLE
Add implementation structs to oscalkit 

### PR DESCRIPTION
**Pull request by minhaj10p**

### CODE REVIEW

**Comment by anweiss** [comment]
This is a WIP PR as it tracks the model being developed at https://github.com/usnistgov/OSCAL/issues/216.

File: generator/generator_test.go [hunk 1]
```diff
@@ -187,7 +193,10 @@ func TestSubControlsMapping(t *testing.T) {
 		},
 	}
 
-	c := (CreateCatalogsFromProfile(&profile))
+	c, err := (CreateCatalogsFromProfile(&profile))
```
Comment (anweiss) [inline generator/generator_test.go]: No need for the extra parenthesis around the function call.

```suggestion
 	c, err := CreateCatalogsFromProfile(&profile)
```

Comment (minhaj10p) [reply generator/generator_test.go]: Done

**Comment by anweiss** [comment]
LGTM

**Pull request merged**